paywall/src/unlock.js/MainWindowHandler.ts

```diff
@@ -61,16 +61,20 @@ export default class MainWindowHandler {
     // dispatching "unlockProtocol" on the main window
     // and
     this.iframes.data.on(PostMessages.LOCKED, () => {
-      this.dispatchEvent('locked')
-      this.setCachedLockedState(true)
-      // Update the user-facing status with locked/unlocked updates
-      this.lockStatus = 'locked'
+      if (this.lockStatus !== 'locked') {
```

Comment:
How would you feel about extracting this method and the other one below? It isnt a block from this shipping but should help us to start wrangling this